### DISCUSSION TITLE
Having trouble spawning items

### DISCUSSION THREAD

**Discussion by tpdenk** [valence-rs/valence#536] [Q&A] [answered]
I'm trying to spawn items (like when the client is dropping an item or destroying a block), but I can't seem to do that.

I'm trying to spawn an `ItemEntityBundle`.

```rust
pub fn drop_loot(
    mut commands: Commands,
    layers: Query<&EntityLayerId>,
    mut drop_events: EventReader<LootDropEvent>,
) {
    drop_events.iter().for_each(|event| {
        let layer = layers.single();
        info!("dropping loot at {:?}", event.position);
        let mut stack = Stack::default();
        stack.count = 1;
        stack.item = ItemKind::Diamond;
        let bundle = ItemEntityBundle {
            item_stack: stack,
            position: Position::new(DVec3::new(
                event.position.x as f64,
                event.position.y as f64,
                event.position.z as f64,
            )),
            layer: *layer,
            ..Default::default()
        };
        commands.spawn(bundle);
    });
}
```

(`LootDropEvent` is being generated upon destroying a block right now. The system shown here is being triggered - I get the log output).

The furthest I've gotten was this, every other experiment around this resulted in just nothing happening.
With this, I get a disconnect because of some nbt data loading issue (and the server seems broken, because clients get the same error message upon connecting until the server restarts, but that's another story I guess).

Can someone point me into the right direction?

**Comment by dyc3** [comment, marked as answer]
This is definitely a bug. I've filed #537 for it.

**Reply by tpdenk** [reply]
I'd keep this discussion open, as I'm not sure whether that's a bug (issue) plus incorrect use (this discussion) or two bugs or any other combination. Feel free to handle as you see fit. @dyc3
And thanks for the quick answer!

**Reply by dyc3** [reply]
Nothing looks wrong with your code to me.

**Reply by NotThatRqd** [reply]
The issue has been closed, so this should be marked as completed now 👍
